### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by Rhaegar121**
In this PR, I have completed the following tasks.

1. Created app.rb file as console app entry-point 
2. Implemented list_book method
3. Implemented list_people method
4. Implemented create_person method
5. Implemented create_book method
6. Implemented create_rental method
7. Implemented list_rentals method
8. Created main.rb with required methods

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    case select_person
+    when 1
+      print 'Has parent permission? [Y/N]: '
+      permission = gets[0].capitalize
+      permission = permission == 'Y'
+      @people << Student.new(nil, age, name, permission)
+    when 2
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(age, specialization, name)
```
Comment (DammyShittu) [inline app.rb]: - [ ] I think your create teacher method mixed up the parameters. The name of the teacher in the screenshot is free, but it is listed as age. Kindly fix that.

  dodiw

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
```
Comment (DammyShittu) [inline app.rb:24]: - [ ] When I list books and people different data is listed also. Kindly remove the items. We only need to list the basic information about the books and people, not the object memory.

  laos
  iisos

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  def list_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals: '
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book: \"#{rental.book.title} by #{rental.book.author}" if rental.person.id == id
+    end
+  end
```
Comment (DammyShittu) [inline app.rb:77]: - [ ] When I list rental it also listed the object info. Kindly fix that.

  idosa

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+def main
+  app = App.new
+
+  puts "Welcome to School Library App!\n\n"
+
+  loop do
+    puts list_options
+    option = gets.chomp.to_i
+    puts option(option, app)
```
Comment (DammyShittu) [inline main.rb]: - [ ] I think the issues of listed unwanted information comes from the line. If you remove the `puts` everything in the app works just fine. You can leave it as this:

  ```
  loop do
    puts list_options
    option = gets.chomp.to_i
    option(option, app)
    ...
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Rhaegar121,

## Highlights
✔️ Basic interface created

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — APPROVED** [review]
Hi @Rhaegar121, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**